### PR TITLE
Fixes for running CI on a BlueField device

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Few small fixed, needed to make CI pass on BlueField SoC

## What
* Restore LD_LIBRARY_PATH after MPI tests. On latest Linux distros, DT_RPATH is replaced by DT_RUNPATH that has lower priority than LD_LIBRARY_PATH. So keeping that env var set later on makes gtest load UCX libraries from install dir rather than from build dir, which fails test_module gtest (since the test module is not installed).
* When using armclang compiler, force using GCC toolchain from module instead of from the local OS. By default, clang tries to take the one which is newer in some (but not all) cases. Without this fix, there are link issues with basic_string default constructor.
* Reduce number of concurrent IB contexts, otherwise device fails to open.

### CODE REVIEW

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -850,8 +851,11 @@ run_mpi_tests() {
 
 		test_malloc_hooks_mpi
 
-		make_clean distclean
+		# Restore LD_LIBRARY_PATH so subsequent tests will not take UCX libs
+		# from installation directory
+		LD_LIBRARY_PATH=${save_LD_LIBRARY_PATH}
```
Comment (ivankochin) [inline contrib/test_jenkins.sh:856]: Shouldn't we export the `LD_LIBRARY_PATH` here to fully revert the changes made above?
```suggestion
		export LD_LIBRARY_PATH=${save_LD_LIBRARY_PATH}
```

**Comment by yosefe** [comment]
fixed with force-push since it's a small PR

**Comment by yosefe** [comment]
@ivankochin is this ok now?

**Review by ivankochin — APPROVED** [review]
